content/shmem_test.tex

```diff
@@ -0,0 +1,53 @@
+\apisummary{
+  Test whether a variable on the local \ac{PE} has changed.
+}
+
+\begin{apidefinition}
+
+\begin{C11synopsis}
+int shmem_test(TYPE *ivar, shmem_cmp cmp, TYPE value);
+\end{C11synopsis}
+where \TYPE{} is one of the point-to-point synchronization types specified by
+Table \ref{p2psynctypes}.
+
+\begin{Csynopsis}
+int shmem_<TYPENAME>_test(TYPE *ivar, shmem_cmp cmp, TYPE value);
+\end{Csynopsis}
+where \TYPE{} is one of the point-to-point synchronization types and has a
+corresponding \TYPENAME{} specified by Table \ref{p2psynctypes}.
+
+\begin{apiarguments}
+
+  \apiargument{OUT}{ivar}{A pointer to a remotely accessible data object.}
+  \apiargument{IN}{cmp}{The comparison operator that compares \VAR{ivar} with
+    \VAR{value}.}
+  \apiargument{IN}{value}{The value against which the object pointed to
+    by \VAR{ivar} will be compared.}
+
+\end{apiarguments}
+
+\apidescription{
+  \FUNC{shmem\_test} tests the numeric comparison of the symmetric object
+  pointed to by \VAR{ivar} with the value \VAR{value} according to the
+  comparison operator \VAR{cmp}.
+}
+
+\apireturnvalues{
+  \FUNC{shmem\_test} returns 1 if the comparison of the symmetric object
+  pointed to by \VAR{ivar} with the value \VAR{value} according to the
+  comparison operator \VAR{cmp} evalutes to true; otherwise, it returns 0.
+}
+
+\apinotes{
```

Comment:
It would be helpful to add a clarification about sync logic, similar to the one that we have in shmem_wait:

```
The \FUNC{shmem\_test} routine can be used to
portably ensure that memory access operations observe remote updates in the order
enforced by the initiator PE using the \FUNC{shmem\_fence} and \FUNC{shmem\_quiet} routines.
```